### PR TITLE
feat(useCssVar): introduce `observe` option

### LINKED ISSUE #2799: useCssVar - css variables out of sync
### Describe the bug

When I use useCssVar to get css variables, I can get the value of the variable, but if I use setProperty to modify the value of the variable, then the variable value obtained by useCssVar will not change.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-tkmlit?file=src%2FApp.vue&terminal=dev

### System Info

```Shell
all
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by Alfred-Skyblue**
> **Warning**: **⚠️ Slowing down new functions**
>
> Due to the growing audience of VueUse, we received a huge amount of feature requests and pull requests. It's become harder and harder and recently a bit beyond our capacity to maintain the project. In the near future, **we could like slowing down on accepting new features and prioritize the stability and quality of existing functions. New functions to VueUse may not be accpected**. If you come up some new ideas, we advice you to have them in your codebase first instead of proposing to VueUse. You may iterate them a few time and see how them suite your needs and how them can be generalized. If you **really** believe they are useful to the community, you can create PR with your usercases, we are still happy to hear and discuss. Thank you for your understanding.

### Description

fixed #2799 

Fix css variables out of sync.

I rewrote the setProperty method for HTMLElement, so that the variable.value can be updated when the css variable is set, but I don't know if this is the best solution, or useMutationObserver to monitor DOM property changes, but I think performance will be worse.

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useCssVar/index.ts [hunk 1]
```diff
@@ -27,6 +27,18 @@ export function useCssVar(
...
+  tryOnMounted(() => {
+    if (elRef.value) {
+      const oldSetProperty = elRef.value.style.setProperty
+      elRef.value.style.setProperty = function (...args) {
+        const [key, value] = args
+        if (key === resolveUnref(prop) && value !== variable.value)
+          variable.value = value!
+        return oldSetProperty.apply(this, args)
+      }
+    }
```
Comment (antfu) [inline packages/core/useCssVar/index.ts:39]: This is probably not a good idea to monkey patch the native functions, we either need a way to listen to the change, or we just list as a limitation and discourage the usage of directly setProperty

**Review by antfu — CHANGES_REQUESTED** [review]

File: packages/core/useCssVar/index.ts (hunk 1 above)
Comment (Alfred-Skyblue) [reply packages/core/useCssVar/index.ts:39]: We can't prevent other libraries from using setProperty to modify css var. I saw in the [vue3 source code](https://github.com/vuejs/core/blob/main/packages/runtime-dom/src/helpers/useCssVars.ts) that it uses MutationObserver to monitor changes in css var. Can we also use MutationObserver to monitor?

Comment (antfu) [reply packages/core/useCssVar/index.ts:39]: Yeah we should use `MutationObserver`, and adding a flag to enable the watcher so ppl don't need it do pay the cost

Comment (Alfred-Skyblue) [reply packages/core/useCssVar/index.ts]: I added a sync option to enable MutationObserver, which is disabled by default .

**Review by antfu — APPROVED** [review]

**Pull request merged**